inference-engine/src/auto_plugin/auto_plugin.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <map>
+#include <vector>
+#include <string>
+#include <unordered_set>
+
+#include <cpp_interfaces/impl/ie_plugin_internal.hpp>
+#include <cpp_interfaces/interface/ie_internal_plugin_config.hpp>
+#include "auto_exec_network.hpp"
+
+namespace AutoPlugin {
+namespace IE = InferenceEngine;
+using ConfigType = std::map<std::string, std::string>;
+
+class AutoInferencePlugin : public IE::InferencePluginInternal {
+public:
+    AutoInferencePlugin();
+    ~AutoInferencePlugin() = default;
+
+    IE::ExecutableNetworkInternal::Ptr LoadExeNetworkImpl(const IE::CNNNetwork& network, const ConfigType& config) override;
+    IE::QueryNetworkResult QueryNetwork(const IE::CNNNetwork& network, const ConfigType& config) const override;
+    IE::Parameter GetMetric(const std::string& name, const std::map<std::string, IE::Parameter>& options) const override;
+    IE::Parameter GetConfig(const std::string& name, const std::map<std::string, IE::Parameter> & options) const override;
+    void SetConfig(const ConfigType& config) override;
+
+private:
+    std::string GetDeviceChoice() const;
```

Comment:
shall the return type of the GetDeviceChoice be a DeviceName, or maybe be better a vector<DevicenMe> so you don't need to parse them again

Reply:
Yes, already changed, thanks!